control/control.go

```diff
@@ -90,8 +93,34 @@ type pluginControl struct {
 	grpcServer  *grpc.Server
 	closingChan chan bool
 	wg          sync.WaitGroup
+
+	// taskView mutex
+	view *sync.Mutex
+	// taskView is used to keep mapping between taskid and taskData (plugins, metrics, requests and config tree)
+	taskView map[string]*taskData
 }
 
+// taskData stores all task related artifacts
+type taskData struct {
+	// requests are RequestedMetrics which are "raw" metrics from workflow
+	requested []core.RequestedMetric
+	// results are Metrics expanded from requested
+	results    []core.Metric
+	configTree *cdata.ConfigDataTree
+	// plugins are collectors currently tied to results (expanded metrics)
+	plugins []core.Plugin
+}
+
+type corePlugin struct {
```

Comment:
Having both `core.Plugin` and `corePlugin` be so similar seems like it could be a problem. Is there not a struct already defined in control that implements core.Plugin that could be used?